python/lsst/summit/utils/plotting.py

```diff
@@ -0,0 +1,305 @@
+        By default, compass will be placed at pixel (x,y) = (300,300).
+    addLegend : `bool', optional
+       Option to add legend to the plot. Recommended if centroids come from
+       different sources. Default value is False.
+    savePlotAs : `str`, optional
+        The name of the file to save the plot as, including the file extension.
+        The extention must be supported by `matplotlib.pyplot`.
+        If None (default) plot will not be saved.
+    logger : `logging.Logger`, optional
+        The logger to use for errors, created if not supplied.
+    Returns
+    -------
+    figure : `matplotlib.figure.Figure`
+        The rendered image.
+    """
+
+    if not figure:
+        figure = plt.figure(figsize=(10, 10))
+
+    ax = figure.add_subplot(111)
+
+    if not logger:
+        logger = logging.getLogger(__name__)
+
+    match inputData:
+        case np.ndarray():
+            imageData = inputData
+        case afwImage.MaskedImage():
+            imageData = inputData.image.array
+        case afwImage.Image():
+            imageData = inputData.array
+        case afwImage.Exposure():
+            imageData = inputData.image.array
+        case _:
+            raise TypeError("This function accepts numpy array, lsst.afw.image.Exposure components."
+                  " Got {type(inputData)}")
+
+    interval = vis.PercentileInterval(percentile)
+    match stretch:
+        case 'ccs':
+            quantiles = getQuantiles(imageData, 256)
+            norm = colors.BoundaryNorm(quantiles, 256)
+        case 'asinh':
+            norm = vis.ImageNormalize(imageData,
+                                      interval=interval,
+                                      stretch=vis.AsinhStretch(a=0.1))
+        case 'power':
+            norm = vis.ImageNormalize(imageData,
+                                      interval=interval,
+                                      stretch=vis.PowerStretch(a=2))
+        case 'log':
+            norm = vis.ImageNormalize(imageData,
+                                      interval=interval,
+                                      stretch=vis.LogStretch(a=1))
+        case 'linear':
+            norm = vis.ImageNormalize(imageData,
+                                      interval=interval,
+                                      stretch=vis.LinearStretch())
+        case 'sqrt':
+            norm = vis.ImageNormalize(imageData,
+                                      interval=interval,
+                                      stretch=vis.SqrtStretch())
+        case _:
+            raise ValueError(f"Invalid value for stretch : {stretch}. "
+                             "Accepted options are: ccs, asinh, power, log, linear, sqrt.")
+
+    im = ax.imshow(imageData, cmap=cmap, origin='lower', norm=norm, aspect='equal')
+    div = make_axes_locatable(ax)
+    cax = div.append_axes("right", size="5%", pad=0.05)
+    figure.colorbar(im, cax=cax, label="Counts")
```

Comment:
Remove the `Counts` from the colorbar label, as we don't know if that's true or not (might be ADU, might be electrons)